src/api/routes/transcripts.ts

```diff
@@ -52,8 +59,74 @@ const list = procedure
   return g.transcripts;
 });
 
+const getNameMap = procedure
+  .use(authUser())
+  .input(z.object({ transcriptId: z.string() }))
+  .output(z.record(z.string()))
+  .query(async ({ input }) => { return await extractAndGetNameMap(input.transcriptId); });
+
+// expected input should an object of {[handlebarNames]: userIds]}
+const updateNameMap = procedure
+  .use(authUser())
+  .input(z.record(z.string()))
+  .mutation(async ({ input: nameMap }) => 
+{
+  // Construct an array of objects to upsert multiple rows the same time
+  const upsertArray = Object.entries(nameMap)
+    .map(([handlebarName, userId]) => ({
+      handlebarName,
+      userId,
+    }));
+
+  await db.SummaryNameMapping.bulkCreate(upsertArray, { updateOnDuplicate: ['userId'] });
+});
 
 export default router({
   get,
   list,
+  getNameMap,
+  updateNameMap
 });
+
+export async function extractAndGetNameMap(transcriptId: string): Promise<Record<string, string>> {
+  const summaries = await db.Summary.findAll({
+    where: { transcriptId },
+    attributes: summaryAttributes,
+  });
+
+  // find all handlebar from summaries under one transcript
+  let handlebarsSet = new Set<string>();
+
+  for (let s of summaries) {
+    const matches = s.summary.match(/{{(.*?)}}/g);
+    if (matches) {
+      // trim is to remove curly brackets wrapped around handlebars found by regex
```

Comment:
`slice(2, -2)` already removes the brackets. I still don't see why trim is necessary.